plugin/src/main/java/io/fabric8/maven/plugin/mojo/build/ResourceMojo.java

```diff
@@ -293,10 +278,8 @@
     // Access for creating OpenShift binary builds
     private ClusterAccess clusterAccess;
 
-    private PlatformMode platformMode;
+    private RuntimeMode runtimeMode;
```

Comment:
There must be no reference to the runtimeMode in ResourceMojo. Do you know why this is needed ?

Reply:
It's used to find out which runtime we're using here and setting some properties(Maybe they were relevant in fabric8 context): https://github.com/fabric8io/fabric8-maven-plugin/blob/19c368936e53a88d402ff24c553855add5313a85/plugin/src/main/java/io/fabric8/maven/plugin/mojo/build/ResourceMojo.java#L489

Reply:
Plus we add ImageChange triggers in deployment config based on runtime mode: https://github.com/fabric8io/fabric8-maven-plugin/blob/19c368936e53a88d402ff24c553855add5313a85/plugin/src/main/java/io/fabric8/maven/plugin/converter/DeploymentOpenShiftConverter.java#L117-L127

Reply:
We should really try hard to get rid of this. I.e. we should check who is using this property and seek for a solution which doesn't involve this runtime mode.

The reason why I'm arguing so hard for removing it is that it can't be that the resources are created differently depending on whether you are connected to a Kuberntes or OpenShift cluster. This totally defeats repeatable builds.

Reply:
IMO the DeploymentOpenShiftConverter should either use the new PlatformMode or we should question why it adds the image trigger here. If this is because e.g. needed for 'fabric8:build' triggering a redeployment, then this functionality should go to 'fabric8:build' to patch existing deployment configs, or even trigger a redeployment on their own.